pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -76,6 +76,13 @@ func NewServiceDiscovery(configController model.ConfigStoreCache, store model.Is
 				}
 			}
 
+			// If service entry is deleted, cleanup endpoint shards for services.
+			if event == model.EventDelete {
+				for _, svc := range cs {
+					c.XdsUpdater.SvcUpdate(c.Cluster(), string(svc.Hostname), svc.Attributes.Namespace, event)
+				}
```

Comment:
return on delete?

Reply:
It should trigger full push to delete these clusters and push to Envoys. `SvcUpdate` just deletes the shards. right?

Reply:
ah, i missed it